crates/ptrace_probe/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use clap::Parser;
+use composer::api::{Client, Event};
+use eyre::Result;
+use nix::{
+    sys::{ptrace, wait::waitpid},
+    unistd::Pid,
+};
+use syscalls::Sysno;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// PID to attach to and trace.
+    pid: u32,
+
+    /// Server address to receive events.
+    address: Option<String>,
+}
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+    let args = Args::parse();
+    let pid = Pid::from_raw(args.pid as i32);
+
+    let client = args
+        .address
+        .map(Client::new)
+        .unwrap_or_else(Client::try_default)?;
```

Comment:
I tend to like these functional-style pipeline-like constructs, but here (and it may be just my lack of rust proficiency) I find it confusing. It looks like that when `Client::new` fails, we fallback to `Client::try_default` but we definitely want to abort in that case.

Only after opening this in IDE with type hints I realized it works properly. 

I find what @skywhale [used](https://github.com/tonarino/acoustic_profiler/pull/22/files#diff-1e4ce9864165105480b1843623dfa14a41c6731ccc44584283d385a75b1776b7R21-R24) in his probe more readable:

```suggestion
    let client = match args.address {
        Some(address) => Client::new(address)?,
        None => Client::try_default()?,
    };
```

But not a strong opinion here. .)

Reply:
Agreed :) to be honest, even though I use these functional chains a lot, this one is pretty hard to follow, and I also had to double check to make sure the unwrapping order was correct as I was writing it.

Will go with the match alternative.